src/analyses/extractPthread.ml

```diff
@@ -146,12 +144,10 @@ module Tbls = struct
 
     let get k = Hashtbl.find table k
 
-    let get_key v = table |> Hashtbl.enum |> List.of_enum |> List.assoc_inv v
+    let get_key v = table |> Hashtbl.bindings |> List.assoc_inv v
   end
 
-  let all_keys_count table =
-    table |> Hashtbl.keys |> List.of_enum |> List.length
-
+  let all_keys_count table = table |> Hashtbl.to_seq_keys |> Seq.length
```

Comment:
Wouldn't this just be `Hashtbl.length` or something?

Reply:
`Hashtbl.length` documentation says:
> `Hashtbl.length tbl` returns the number of bindings in `tbl`. It takes constant time. Multiple bindings are counted once each, so `Hashtbl.length` gives the number of times `Hashtbl.iter` calls its first argument.

But I am not entirely sure that the table here always has only one value per key.